doc/reference/kernel/synchronization/condvar.rst

```diff
@@ -0,0 +1,139 @@
+.. _condvar:
+
+Condition Variables
+####################
+
+A :dfn:`condition variable` Condition variables is a synchronization primitive
+that enables threads to wait until a particular condition occurs.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Concepts
+********
+
+Any number of condition variables can be defined (limited only by available RAM). Each
+condition variable is referenced by its memory address.
+
+To wait for a condition to become true, a thread can make use of a condition
+variable.
+
+A condition variable is basically a queue of threads that threads can put
+themselves on when some state of execution (i.e., some condition) is not as
+desired (by waiting on the condition), the function
```

Comment:
`s/, the function/. The function/` would help, but the parentheticals in first fragment still make it a bit hard to understand.